static/src/javascripts/projects/common/modules/experiments/tests/measure-understanding.js

```diff
@@ -0,0 +1,116 @@
+
+    // Ophan survey id
+    const component = 'measure_understanding';
+
+    // Test duration
+    const start = '2017-04-27';
+    const expiry = '2017-05-27';
+
+    // will run in articles only if no story questions atom is already there
+    // *and* the user hasn't already answered the question
+    const canRun = () =>
+        config.page.contentType === 'Article' &&
+        !config.page.isAdvertisementFeature &&
+        !localStorage.get(id) &&
+        !document.getElementsByClassName('js-view-tracking-component').length;
+
+    // at the click of a button, we send a notification to ophan and clean up
+    // registered listeners to prevent duplicate calls
+    const onClick = (evt: Event) => {
+        // #? Flow will complain if we don't type check
+        if (!(evt.target instanceof Element)) {
+            throw new Error('This will never happen');
+        }
+        const button = evt.target.closest('.js-button');
+        const value = button && button.getAttribute('data-value');
+        if (value) {
+            // #? Flow will complain if we don't type check
+            if (!(evt.currentTarget instanceof Element)) {
+                throw new Error('This will never happen');
+            }
+            const question = evt.currentTarget;
+            const feedback = question.lastElementChild;
+            question.removeEventListener('click', onClick);
+            fastdom
+                .write(() => {
+                    question.classList.add('is-answered');
+                    [...question.querySelectorAll('.js-button')].forEach(b =>
+                        b.setAttribute('hidden', 'hidden')
+                    );
+                    if (feedback) {
+                        feedback.removeAttribute('hidden');
+                    }
+                })
+                .then(() => {
+                    localStorage.set(id, true, { expires: Date.parse(expiry) });
+                    ophan.record({ component, value });
+                    mediator.emit('ab:understanding:complete');
+                });
+        }
+    };
+
+    // test bootstrap: runs when the user is in the corresponding variant
+    const test = () => {
+        const html = template(measureUnderstandingStr, {
+            thumbsUp: thumb,
+            thumbsDown: thumb,
+        });
+        const articleBody = document.getElementsByClassName('js-article__body')[
+            0
+        ];
+        fastdom
+            .write(() => {
+                articleBody.insertAdjacentHTML('beforeend', html);
+                return articleBody.lastElementChild;
+            })
+            .then(clicktrap => {
+                addEventListener(clicktrap, 'click', onClick);
+                mediator.emit('ab:understanding:displayed');
+            });
+    };
+
+    // registers an impression (in our case, when the form is added to the dom)
+    const impression = track => {
+        mediator.on('ab:understanding:displayed', track);
```

Comment:
I've had a read about Mediator, but could you please expand on this a little? I'm not sure I know what these consts are doing  🤔